tdms/tests/system/utils.py

```diff
@@ -178,3 +178,22 @@ def download_data(url: str, to: Path) -> None:
     os.chdir(cwd)
 
     return None
+
+
+def compare_output(input_filename, reference_output_filename):
+    """
+    Run TDMS using `input_filename`, then compare the output to the data
+    saved in `reference_output_filename`.
+
+    Checks that the output .mat file (with a HDF5 format) contains tensors with
+    relative mean square values within numerical precision of the reference.
+    """
+    output_filename = "pstd_output.mat"
+    run_tdms(input_filename, output_filename)
+
+    reference = HDF5File(reference_output_filename)
+    output = HDF5File(output_filename)
+    try:
+        assert output.matches(reference)
```

Comment:
Should we have a (maybe optional) assert message passed all the way in here? Or do you think that's overkill?

It's just: I see "blah doesn't match reference". And it's not clear from the error if I've broken the steady-state or the extrapolation or what.

Reply:
I've done two things to improve this:
1. Change the `matches` method to `assert_matches`
2. Used numpy's built in `np.testing.assert_equal` to check the data, which will print useful information if the comparison fails somehow